test/solarwinds_apm/config_test.rb

```diff
@@ -0,0 +1,274 @@
+    it 'with config env true' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with config env false' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, false, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong config env, use default false' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, 'true', false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong config env, use default true' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, 'true', true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with nil env, config env true, use default false, should be true' do
+      SolarWindsAPM::Config.enable_disable_config(nil, :dummy_key, true, false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with nil env, empty config env, use default false, should be false' do
+      SolarWindsAPM::Config.enable_disable_config(nil, :dummy_key, '', false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with nil env, empty config env, use default true, should be true' do
+      SolarWindsAPM::Config.enable_disable_config(nil, :dummy_key, '', true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+  end
+
+  describe '#set_log_level' do
+    before do
+      SolarWindsAPM::Config.initialize
+      @current_debug_level = ENV['SW_APM_DEBUG_LEVEL']
+      ENV.delete('SW_APM_DEBUG_LEVEL')
+    end
+
+    after do
+      ENV['SW_APM_DEBUG_LEVEL'] = @current_debug_level
+    end
+    
+    it 'debug_level is out of range use default 1' do
+      SolarWindsAPM::Config[:debug_level] = 7
+      SolarWindsAPM::Config.set_log_level
+      _(SolarWindsAPM.logger.level).must_equal 1
+    end
+
+    it 'debug_level is in the range' do
+      SolarWindsAPM::Config[:debug_level] = 1
+      SolarWindsAPM::Config.set_log_level
+      _(SolarWindsAPM.logger.level).must_equal 3
+    end
+
+    it 'env var override config' do
+      ENV['SW_APM_DEBUG_LEVEL'] = '3'
+      SolarWindsAPM::Config[:debug_level] = 1
+      SolarWindsAPM::Config.set_log_level
+      _(SolarWindsAPM.logger.level).must_equal 1
+      ENV.delete('SW_APM_DEBUG_LEVEL')
+    end
+
+    # because of [4 - debug_level, 0].max, if SW_APM_DEBUG_LEVEL is out of range then, it will set to 0
+    it 'env var override config but out of range' do
+      ENV['SW_APM_DEBUG_LEVEL'] = '7'
+      SolarWindsAPM::Config[:debug_level] = 1
+      SolarWindsAPM::Config.set_log_level
+      _(SolarWindsAPM.logger.level).must_equal 0
+      ENV.delete('SW_APM_DEBUG_LEVEL')
+    end
+
+    it 'env var override config but out of range by less than 0' do
```

Comment:
I am honestly very confused and mystified by the range verification... you might be too based on the comment in line 217? Can this logic be cleaned up to actually make sense?

Reply:
Yes, inherited `set_log_level` function is kind confusing. I will make it clear